src/DataViews/PresetsDataViewTest.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_split.h>
+#include <gmock/gmock-actions.h>
+#include <gmock/gmock-cardinalities.h>
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "DataViews/DataView.h"
+#include "DataViews/PresetLoadState.h"
+#include "DataViews/PresetsDataView.h"
+#include "MetricsUploader/MetricsUploaderStub.h"
+#include "MockAppInterface.h"
+#include "OrbitBase/File.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/TemporaryFile.h"
+#include "OrbitBase/TestUtils.h"
+#include "PresetFile/PresetFile.h"
+#include "preset.pb.h"
+namespace {
+// This is just a helper type to handle colors. Note that Color from `OrbitGl/CoreMath.h` is not
+// available in this module.
+struct Color {
+  unsigned char r, g, b;
+  friend bool operator==(const Color& lhs, const Color& rhs) {
+    return std::tie(lhs.r, lhs.g, lhs.b) == std::tie(rhs.r, rhs.g, rhs.b);
+  }
+  friend bool operator!=(const Color& lhs, const Color& rhs) { return !(lhs == rhs); }
+};
+}  // namespace
+
+TEST(PresetsDataView, Columns) {
+  orbit_metrics_uploader::MetricsUploaderStub metrics_uploader{};
+  orbit_data_views::MockAppInterface app{};
+  orbit_data_views::PresetsDataView view{&app, &metrics_uploader};
+
+  EXPECT_GE(view.GetColumns().size(), 1);
+  for (const auto& column : view.GetColumns()) {
+    EXPECT_FALSE(column.header.empty());
+  }
+}
+
+TEST(PresetsDataView, Empty) {
+  orbit_metrics_uploader::MetricsUploaderStub metrics_uploader{};
+  orbit_data_views::MockAppInterface app{};
+  orbit_data_views::PresetsDataView view{&app, &metrics_uploader};
+
+  EXPECT_EQ(view.GetNumElements(), 0);
+  EXPECT_EQ(view.GetLabel(), "Presets");
+}
+
+TEST(PresetsDataView, LoadState) {
+  orbit_metrics_uploader::MetricsUploaderStub metrics_uploader{};
+  orbit_data_views::MockAppInterface app{};
+  orbit_data_views::PresetsDataView view{&app, &metrics_uploader};
+
+  // GetPresetLoadState is called once per `GetValue` and `GetToolTip` call.
+  auto load_state = orbit_data_views::PresetLoadState::kLoadable;
+  EXPECT_CALL(app, GetPresetLoadState).Times(9).WillRepeatedly(testing::ReturnPointee(&load_state));
```

Comment:
Do we want to fix the number of times this method called in the test? You are using AnyNumber in other tests...

Reply:
This was very much intentional. Since this test is about the preset load state I check the number of calls here. But I won't check it in the other tests since there it's not under test. If somethings breaks around `GetPresetLoadState` ideally only this test fails and it will be easy to pinpoint the problem. Does that make sense?